internal/controller/streaming_manager.go

```diff
@@ -32,6 +34,18 @@ type streamManager struct {
 	logger   *zap.SugaredLogger
 }
 
+type ClientConnWrapper struct {
+	conn *grpc.ClientConn
+}
+
+func (w *ClientConnWrapper) GetState() connectivity.State {
+	return w.conn.GetState()
+}
+
+func (w *ClientConnWrapper) Close() error {
+	return w.conn.Close()
+}
+
```

Comment:
Remove. You should already be able to convert a `ClientConn` to a `ClientConnInterface`.